Quetoo.vs15/src/config.h

```diff
@@ -0,0 +1,86 @@
+#ifndef __attribute__
+#define __attribute__(...)
+#endif
+
+#ifndef __thread
+#define __thread __declspec(thread)
+#endif
+
+#ifndef usleep
+#define usleep(t) Sleep(t / 1000)
+#endif
+
+#ifndef S_ISDIR
+#define S_ISDIR(m) (((m) & S_IFDIR) == S_IFDIR)
+#endif

```

Comment:
MSVC has stat.h, it just doesn't include S_ISDIR for some reason. I don't think I can cascade the includes either (as in, include MSVC' stat.h from our stat.h), which is why I didn't make a stat.h.

unistd is a slight exception; I specifically designed this PR to not require code modifications to the third party libraries. I added unistd so that one of the libraries which tries to include it won't fail (and will get ssize_t, which is why it requested unistd in the first place.. on MSVC, size_t is implicitly defined by the compiler but ssize_t does not exist. Go figure :) )